astro_reduction_tool.py

```diff
@@ -0,0 +1,222 @@
+            len(file_list.data_list)
+        )
+    return file_list.data_list
+
+
+def master_flat_generator(flat_p, master_dark, master_bias, filt):
+    m_flat = []
+    m_flat_n = []
+    keys = ["imagetyp", "filter"]
+    for f in filt:
+        flat_data = read_path_into_CCDData(flat_p, keys, f)
+        flat_subtracted_list = debias_dedark(flat_data, master_dark, master_bias)
+        combined_flat = ccdp.combine(
+            flat_subtracted_list,
+            method="median",
+            unit=u.adu,
+            scale=inv_median,
+            sigma_clip=True,
+            sigma_clip_low_thresh=5.0,
+            sigma_clip_high_thresh=5.0,
+            sigma_clip_func=np.ma.median,
+            sigma_clip_dev_func=mad_std,
+            flat=2,
+        )
+        combined_flat.meta["combined"] = True
+        name = os.path.join(flat_p,"master_flat_{}.fits".format(f))
+        combined_flat.write(name, overwrite=True)
+        now = datetime.now()
+        formatted_time = now.strftime("[%Y-%m-%dT%H:%M:%S.%f]")
+        logging.info("%s  Created master_flat in %s filtr", formatted_time, f)
+        m_flat.append(combined_flat)
+        m_flat_n.append(name)
+        m_flat_list = ccd_data(m_flat,m_flat_n)
+    return m_flat_list
+
+
+def folders_reduction(dir_folder, m_dark, m_bias, m_flat_dictionary, d):
+    path = os.path.join(dir_folder,d)
+    ifc_folder = ccdp.ImageFileCollection(path, keywords=["imagetyp", "filter"])
+    filt = set(h["filter"] for h in ifc_folder.headers())
+    if os.path.exists(os.path.join(path, "pipeline_out")) is not True:
+        os.mkdir(os.path.join(path, "pipeline_out"))
+    for f in filt:
+        ccd_folder = read_path_into_CCDData(
+            path, ["imagetyp", "filter"], f
+        )
+        max = len(ccd_folder.data_list)
+        if max != 0:
+            x=0
+            for x in range(0,len(ccd_folder.data_list)):
+                ccd_name = ccd_folder.name_list[x]
+                ccd_file = ccd_folder.data_list[x]
+                bias_subtracted = ccdp.subtract_bias(ccd_file, m_bias)
+                dark_subtracted = ccdp.subtract_dark(
+                    bias_subtracted,
+                    m_dark,
+                    exposure_time="EXPTIME",
+                    exposure_unit=u.adu,
+                    scale=True,
+                )
+                ccd_file = ccdp.flat_correct(dark_subtracted, m_flat_dictionary[f],min_value=0.01)
+                ccd_file.write(os.path.join(path,"pipeline_out", ccd_name), overwrite=True)
+                now = datetime.now()
+                formatted_time = now.strftime("[%Y-%m-%dT%H:%M:%S.%f]")
+                logging.info("%s  Saved %s (%d in %d)", formatted_time, os.path.join(path,"pipeline_out", ccd_name),x,max)
+            now = datetime.now()
+            formatted_time = now.strftime("[%Y-%m-%dT%H:%M:%S.%f]")
+            logging.info("%s  Subtracted %s folder in %s filtr", formatted_time, d, f)
+
+
+if __name__ == "__main__":
```

Comment:
Dla czytelnosci przenioslbym calosc kodu z sekcji ponizej do osobnej funkcji `main()` a na koncu pliku zostawil tylko:
```suggestion
if __name__ == "__main__":
    main()
```

Tak bedzie badziej naturalnie.

Dodatkowo z funkcji `main()` wydzielilbym do osobnej funkcji sekcje parsowania argumentow (nazwijmy ja `parse_args()` i dalsza czesc (nazwijmy ja `astro_reduction()`), dzieki czemu `main()` moglby wygladac tak:

```
def main():
    args = parse_args()
    astro_reduction(args)
```

Wszystko dla czytelnosci kodu...